lib/db.c

```diff
@@ -8818,9 +8818,47 @@ grn_obj_set_info_source_validate(grn_ctx *ctx, grn_obj *obj, grn_obj *value)
   if (!lexicon_domain) {
     goto exit;
   }
+  lexicon_domain_is_table = grn_obj_is_table(ctx, lexicon_domain);
+  {
+    grn_obj *tokenizer;
+    grn_table_get_info(ctx, lexicon, NULL, NULL, &tokenizer, NULL, NULL);
+    lexicon_have_tokenizer = (tokenizer != NULL);
+  }
 
   source_ids = (grn_id *)GRN_BULK_HEAD(value);
   n_source_ids = GRN_BULK_VSIZE(value) / sizeof(grn_id);
+  if (grn_obj_is_index_column(ctx, obj)) {
+    if (n_source_ids == 1) {
+      grn_obj *source;
+
+      source = grn_ctx_at(ctx, source_ids[0]);
+      switch (source->header.type) {
```

Comment:
Can we use `grn_obj_is_vector_column()`?

Reply:
Thank you for your comment.
I've fixed it in 197d92f.